internal/services/network/virtual_hub_resource.go

```diff
@@ -131,6 +131,21 @@ func resourceVirtualHub() *pluginsdk.Resource {
 				Type:     pluginsdk.TypeString,
 				Computed: true,
 			},
+
+			"virtual_router_auto_scale_configuration": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"min_capacity": {
+							Type:         pluginsdk.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntAtLeast(2),
+						},
+					},
+				},
+			},
```

Comment:
Given that there is only one field in this block, it may simplify things to flatten this instead?
```suggestion
"virtual_router_auto_scale_min_capacity": {
				Type:         pluginsdk.TypeInt,
				Optional:     true,
				ValidateFunc: validation.IntBetween(2, 50),
			},
```

Reply:
Can do, however my impression is that this is a block in the [REST API](https://learn.microsoft.com/en-us/rest/api/virtualwan/virtual-hubs/create-or-update?tabs=HTTP#virtualrouterautoscaleconfiguration) because they plan on adding more fields. Should I still flatten it at the risk of making things more complicated later on?

Reply:
@prototypicalpro - for now we can flatten it because it could also not have any new properties added, and if there are its fairly easy to add the block in and deprecate the property gracefully

Reply:
Sounds good, I've made the change.